File: apps/android/app/src/main/java/com/example/shop/feature/auth/data/AuthApi.kt
package com.example.shop.feature.auth.data

import com.example.shop.core.model.ApiResponse
import com.example.shop.feature.auth.data.model.AuthResponse
import com.example.shop.feature.auth.data.model.LoginRequest
import com.example.shop.feature.auth.data.model.LogoutRequest
import com.example.shop.feature.auth.data.model.RegisterRequest
import kotlinx.serialization.json.JsonElement
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthApi {
    @POST("api/v1/auth/login")
    suspend fun login(@Body request: LoginRequest): ApiResponse<AuthResponse>

    @POST("api/v1/auth/register")
    suspend fun register(@Body request: RegisterRequest): ApiResponse<AuthResponse>

    @POST("api/v1/auth/logout")
    suspend fun logout(@Body request: LogoutRequest): ApiResponse<JsonElement?>
}

File: apps/ios/Shop/Feature/Auth/Data/AuthAPI.swift
import Foundation

enum AuthAPI {
    static func login(email: String, password: String) async throws -> AuthResponse {
        try await APIClient.shared.request(
            .authLogin,
            body: LoginRequest(email: email, password: password)
        )
    }

    static func register(email: String, password: String, nickname: String?) async throws -> AuthResponse {
        try await APIClient.shared.request(
            .authRegister,
            body: RegisterRequest(email: email, password: password, nickname: nickname)
        )
    }

    static func logout(refreshToken: String?) async throws {
        try await APIClient.shared.requestVoid(
            .authLogout,
            body: LogoutRequest(refreshToken: refreshToken),
            needsAuth: true
        )
    }
}
